cli/src/stake.rs

```diff
@@ -1892,14 +1921,26 @@ pub fn process_split_stake(
             return Err(CliError::BadParameter(err_msg).into());
         }
 
-        let minimum_balance =
-            rpc_client.get_minimum_balance_for_rent_exemption(StakeState::size_of())?;
+        if !force {
+            let minimum_balance = rpc_client
+                .get_minimum_balance_for_rent_exemption(StakeState::size_of())?
+                + rpc_client.get_stake_minimum_delegation()?;
+            if lamports < minimum_balance {
+                return Err(CliError::BadParameter(format!(
+                    "need at least {minimum_balance} lamports for new stake account to be rent exempt and meet minimum delegation, provided lamports: {lamports}"
+                ))
+                .into());
+            }
 
-        if lamports < minimum_balance {
-            return Err(CliError::BadParameter(format!(
-                "need at least {minimum_balance} lamports for stake account to be rent exempt, provided lamports: {lamports}"
-            ))
-            .into());
+            let source_post_balance = rpc_client
+                .get_balance(stake_account_pubkey)?
+                .saturating_sub(lamports);
+            if source_post_balance < minimum_balance {
```

Comment:
This check isn't necessarily valid in all cases either, in case the source stake account has additional undelegated lamports on it. You'd have to fetch the lamports and the stake activation to figure out if it has enough.

But to be honest, that situation is a bit rarer, so maybe the person can just throw in `--skip-balance-checks` and enjoy the rest of their day :sweat_smile:

Reply:
Yeah getting stake activation adds a lot more complexity, I'd rather let it fail then and have the user use `--skip-balance-checks`, the CLI is a client-tool, and not strictly a representation of what will work or not work on-chain and as such should account for 99% of use cases in guiding the user, imho. The payoff for adding the stake activation here in terms of complexity and time taken to compute for the user seems poor here

Reply:
Yeah no problem, fine by me!